### PR TITLE
hotplug_unplug_nic_with_viommu:Hotplug/unplug nic based on viommu device

### PR DESCRIPTION
**Pull request by yanglei-rh**
Hotplug/unplug nic with iommu_platform=on based on virtio iommu device

ID:2148348
Signed-off-by: Lei Yang <leiayang@redhat.com>

### CODE REVIEW

**Comment by yanglei-rh** [comment]
Test Pass:
(1/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.hotplug_unplug_with_viommu.direct_plug.q35: PASS (296.60 s)
(2/2) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.hotplug_unplug_with_viommu.non_direct_plug.q35: PASS (287.75 s)

**Comment by yanglei-rh** [comment]
Hello @PaulYuuu Could you please help me review this patch? Thanks a lot.

File: qemu/tests/cfg/hotplug_unplug_with_viommu.cfg [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    only Linux
+    only virtio_net
+    login_timeout = 360
+    clone_master = yes
+    master_images_clone = image1
+    remove_image_image1 = yes
+    repeat_times = 1
+    virtio_iommu = yes
+    pci_model = virtio-net-pci
+    netdev_extra_params_hotplug_nic1 = ",vhost=on"
```
Comment (PaulYuuu) [inline qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:15]: `vhost = on`?

File: qemu/tests/cfg/hotplug_unplug_with_viommu.cfg [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    nic_extra_params_hotplug_nic1 = ",iommu_platform=on"
+    variants:
+        - direct_plug:
+              virtio_iommu_direct_plug = yes
+        - non_direct_plug:
+              virtio_iommu_direct_plug = no
```
Comment (PaulYuuu) [inline qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:21]:
```suggestion
```

File: qemu/tests/cfg/hotplug_unplug_with_viommu.cfg [hunk 3]
```diff
@@ -0,0 +1,21 @@
+- hotplug_unplug_with_viommu:
+    virt_test_type = qemu
+    type = nic_hotplug
+    required_qemu= [7.0.0,)
+    only q35
```
Comment (PaulYuuu) [inline qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:5]: Need to change.

File: qemu/tests/cfg/hotplug_unplug_with_viommu.cfg (hunk 1 above)
Comment (yanglei-rh) [reply qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:15]: > `vhost = on`?

Let me check it again. I thought that vhost =on would not affect the parameters of hotplug, so I added this line of configuration.

File: qemu/tests/cfg/hotplug_unplug_with_viommu.cfg (hunk 2 above)
Comment (yanglei-rh) [reply qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:21]: This line will be removed after next update.

File: qemu/tests/cfg/hotplug_unplug_with_viommu.cfg (hunk 3 above)
Comment (yanglei-rh) [reply qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:5]: virtio iommu should only be supported on the Q35 platform, isn't it? Please correct me if I'm wrong. Thanks a lot.

Comment (PaulYuuu) [reply qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:5]: https://gitlab.com/qemu-project/qemu/-/blob/master/softmmu/qdev-monitor.c#L59-64
https://gitlab.com/qemu-project/qemu/-/blob/master/softmmu/qdev-monitor.c#L92
https://gitlab.com/libvirt/libvirt/-/blob/master/src/qemu/qemu_validate.c#L4846-4853

Comment (yanglei-rh) [reply qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:5]: Ok, thanks for your help. But I'm not familiar with ARM, how should I specify it?

Comment (yanglei-rh) [reply qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:5]: Hello @PaulYuuu Can `only aarch64 x86_64` meet the requirements?

File: qemu/tests/cfg/hotplug_unplug_with_viommu.cfg (hunk 1 above)
Comment (yanglei-rh) [reply qemu/tests/cfg/hotplug_unplug_with_viommu.cfg:15]: > > `vhost = on`?
> 
> Let me check it again. I thought that vhost =on would not affect the parameters of hotplug, so I added this line of configuration.

@PaulYuuu I tested it without this line, according to test reuslt, if this line be removed, hotplug command will not include vhost. So I think we should keep it.Please correct me if I'm wrong.
`{'execute': 'netdev_add', 'arguments': {'type': 'tap', 'id': 'idRqG4TJ', 'fd': '91'}, 'id': 'YNB1Q184'}`

**Comment by yanglei-rh** [comment]
Test Pass:
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.hotplug_unplug_with_viommu.q35: PASS (283.75 s)

**Comment by yanglei-rh** [comment]
@PaulYuuu Please help me review it again, thanks a lot.

**Review by PaulYuuu — APPROVED** [review]

**Comment by PaulYuuu** [comment]
Please modify the cfg name to include `nic`, such as `hotplug_unplug_nic_with_viommu`.

**Comment by yanglei-rh** [comment]
Test Pass:
 (1/1) Host_RHEL.m9.u2.ovmf.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.9.2.0.x86_64.io-github-autotest-qemu.hotplug_unplug_nic_with_viommu.q35: PASS (265.09 s)

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**